packages/next/server/image-optimizer.ts

```diff
@@ -253,8 +253,17 @@ export async function imageOptimizer(
         mockRes.write = (chunk: Buffer | string) => {
           resBuffers.push(Buffer.isBuffer(chunk) ? chunk : Buffer.from(chunk))
         }
-        mockRes._write = (chunk: Buffer | string) => {
+        mockRes._write = (
+          chunk: Buffer | string,
+          encoding: string,
+          callback: () => void
+        ) => {
           mockRes.write(chunk)
+          // According to Node.js documentation, the callback MUST be invoked to signal that
+          // the write completed successfully. If this callback is not invoked, the 'finish' event
+          // will not be emitted.
+          // https://nodejs.org/docs/latest-v16.x/api/stream.html#writable_writechunk-encoding-callback
+          callback()
```

Comment:
```suggestion
          if (callback) callback()
```

The docs mention that `encoding` and `callback` are both optional parameters so best to check it exists

Reply:
I think the two parameters are optional when calling `write()`
https://nodejs.org/docs/latest-v16.x/api/stream.html#writablewritechunk-encoding-callback
but not when calling `_write()`
https://nodejs.org/docs/latest-v16.x/api/stream.html#writable_writechunk-encoding-callback